qiskit/chemistry/drivers/base_driver.py

```diff
@@ -0,0 +1,93 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+This module implements the abstract base class for driver modules.
+"""
+
+from typing import Optional
+from abc import ABC, abstractmethod
+
+from .molecule import Molecule
+from ..qiskit_chemistry_error import QiskitChemistryError
+
+
+class BaseDriver(ABC):
+    """
+    Base class for Qiskit's chemistry drivers.
+    """
+
+    @abstractmethod
+    def __init__(self,
+                 molecule: Optional[Molecule] = None,
+                 basis: str = 'sto3g',
+                 hf_method: str = 'rhf',
```

Comment:
I think it would make sense to use an `Enum` for this. At least because this is the same way it is done in the `PySCFDriver` and `PyQuanteDriver`.

Reply:
Clearly we have to be careful here - this is the base class of all drivers so if the enums are different (I believe your PySCF changes took things in that direction) then its not going to work. Either way this is the super constructor that is only called by the derived classes. Now PSI4 and G16 drivers were put as string at their level - they could be made an enums. These are the instances that are created - its then up to the driver to convert its enum into the right hf_method string (presently its just the enum value so its all simple). There is a setter at that BaseDriver level that is a string which I guess an external user could call in which case you would need the right string.

Reply:
Well I just thought it could make sense. Obviously there could be complications with the existing `Enum`s for pyscf and pyquante but those could either by removed entirely and a check could be done whether the requested method is supported or the `Enum` could be converted. In the end, its just all strings anyways.
But if you see too many problems I am also fine with leaving it as is.

Reply:
My preference is for enums on the drivers since an IDE etc can help in populating them out via auto-complete. Strings someone has to go look at docstring where hopefully its listed. This base class no-one should be directly constructuting so it should be less of an issue I think.